indy_client/agent/run_agent.py

```diff
@@ -64,7 +64,7 @@ def runAgent(agent, looper=None, bootstrap=None):
 
     def is_connected(agent):
         client = agent.client
-        if not client.can_send_write_requests():
+        if (client.mode is None) or (not client.can_send_write_requests()):
```

Comment:
I think it makes sense to check not defined yet mode inside can_send_... methods.

Reply:
I agree, I just wanted to make a hot fix here. I will send a separate PR to move this condition to can_send_write_requests()

Reply:
i see